azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -167,6 +178,10 @@ protected OperatingSystemEnum getOsEnum() throws MojoExecutionException {
     }
 
     protected ArtifactHandler getArtifactHandler() throws MojoExecutionException {
+        if (getOsEnum() == OperatingSystemEnum.Docker) {
```

Comment:
Plan to create a new deployment type `Docker`, switch by deployment type instead of if in later pr, which contains the auto deployment type judgement for linux/windows function.